lib/http2/hpack.c

```diff
@@ -109,7 +116,11 @@ static int contains_invalid_field_name_char(const char *s, size_t len)
     for (; len != 0; ++s, --len) {
         unsigned char ch = (unsigned char)*s;
         if (!valid_h2_field_name_char[ch]) {
-            return 1;
+            if (ch - 'A' < 26U) {
+                *err_desc = err_found_upper_case_in_header_name;
+                return 1;
+            }
+            *err_desc = soft_err_found_invalid_char_in_header_name;
```

Comment:
> I think you would also need to adjust the return code as well.

Is that so? In the soft error case, we want to continue parsing. Passing the right error code is handled in https://github.com/h2o/h2o/pull/974/files#diff-cdf5c2480fce6f3c41c628a00e404f32R392